### PR TITLE
cpu/samd21: set optimization level to -Os

### PR DESCRIPTION
**Pull request by haukepetersen**
I was wondering about the huge code-size for the samr21-xpro - turned out, the optimization for the board is turned off... Looks better now.

### CODE REVIEW

**Comment by PeterKietzmann** [comment]
Nice one! ACK, waiting for Travis..

**Comment by OlegHahm** [comment]
Have you tested with some applications to check if this has no unexpected side effect?

**Comment by kaspar030** [comment]
We did, it has some problems with the GPIO implementation AFAIR.

**Comment by PeterKietzmann** [comment]
Tested with some easy applications but not with peripheral tests. In this case I revert my ACK and need to do some further testings :(

**Comment by haukepetersen** [comment]
I did some more tests, the GPIO drivers are not cleanly implemented. I was not really going to look into this in detail... But I will try to do so.

Just a remark: the GPIO issues are actually unrelated to this PR, it just seems that enabling the compiler optimization triggers the compiler to detect them...

**Comment by OlegHahm** [comment]
> Just a remark: the GPIO issues are actually unrelated to this PR, it just seems that enabling the compiler optimization triggers the compiler to detect them...

Well, while I think it's okay for this particular test application, that was exactly the idea behind my question: even if the underlying problem is unrelated to this change, it may introduce failures that won't happen without the optimization. rpl_udp is typically a good candidate, but I have only one SAMR21 here.

**Comment by haukepetersen** [comment]
I will look into this once I find some time...

**Comment by PeterKietzmann** [comment]
I think we can either (i) merge this PR and open an issue, close this and open an issue (iii) close this PR as it is also part of #2852 and move the discussion there

**Comment by haukepetersen** [comment]
I say lets leave this PR open for now and and see if after #2865 is merged either this PR or #2852 is faster...

**Comment by haukepetersen** [comment]
Just saw that the GPIO warnings are not the only issue for the `samr21-xpro` board. When enabling `-Os`, the `tests/driver_at86rf2xx` application does also not work -> it returns an error in the radio initialization function. I would suspect something the low level drivers to cause that...

**Comment by bapclenet** [comment]
@haukepetersen, might be the cause of the GPIO as well? or SPI?

**Comment by haukepetersen** [comment]
I had no time to look further into this, yet. But I hope to get to it in the end of the week.

**Comment by haukepetersen** [comment]
Good news: the samr21 is now doing fine with `-Os`. So please verify this and let's merge this soon to save some time flashing the nodes!

File: cpu/samd21/Makefile.include [hunk 1]
```diff
@@ -6,7 +6,4 @@ export CFLAGS += -DDONT_USE_CMSIS_INIT
 # use the hwtimer compatibility layer
 USEMODULE += hwtimer_compat
 
-# TODO: remove once the CPU implementation is stable with -Os
-CFLAGS_OPT = -O0
-
```
Comment (bapclenet) [inline cpu/samd21/Makefile.include]: Shouldn't we change it by -Os then?

Comment (kaspar030) [inline cpu/samd21/Makefile.include]: No, the common makefile defines ```CFLAGS_OPT ?= -Os```.

Comment (bapclenet) [inline cpu/samd21/Makefile.include]: ok

**Comment by haukepetersen** [comment]
somebody up to acking this?

**Comment by kaspar030** [comment]
Just connected the samr21, sec

**Comment by kaspar030** [comment]
works. ACK&go.

**Comment by haukepetersen** [comment]
awesome, thx.

**Comment by haukepetersen** [comment]
seems we are moving in the right direction with this board :-)

**Comment by kaspar030** [comment]
yeah ;)

**Pull request merged**